### PR TITLE
HPCC-15845 Add support for average/sd on query stats

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith probably not quite ready for pulling, but worth reviewing to see if would make life easier in Thor.

**Comment by jakesmith** [comment]
@ghalliday - all looks sensible, couldn't spot any issues.

>life easier in Thor

to better spot skew?

**Comment by ghalliday** [comment]
because your exisiting classes for calculating skew etc. could be replaced.

I'll look now to see what else needs to be done for it to be merged at least.

File: system/jlib/jstats.cpp [hunk 1]
```diff
@@ -1672,6 +1763,114 @@ bool CRuntimeStatisticCollection::serialize(MemoryBuffer& out) const
...
+            StatisticKind kind = getKind(i);
+            StatisticKind serialKind= querySerializedKind(kind);
+
+            unsigned __int64 minValue = convertMeasure(kind, serialKind, cur.min);
+            unsigned __int64 maxValue = convertMeasure(kind, serialKind, cur.max);
+            if (minValue != maxValue)
+            {
+                double sum = (double)convertMeasure(kind, serialKind, values[i].get());
+                //Sum of squares needs to be translated twice
+                double sumSquares = (double)convertMeasure(kind, serialKind, convertMeasure(kind, serialKind, cur.sumSquares));
```
Comment (ghalliday) [inline system/jlib/jstats.cpp]: This implicitly casts from a double and may overflow an int.  It needs a different function.

**Comment by ghalliday** [comment]
@richardkchapman I think this is ok to merge.  Most of the new code isn't used yet - it requires some work in roxie for the summaries to use the new class.  I think it will mean that function timings are included in the summary stats, but nothing else.
I don't mind if it is merged, or wait until I get back.

**Pull request merged**